### PR TITLE
[BugFix] Fix parameters arithmetic.

### LINKED ISSUE #242: Fix parameters arithmetic
While trying to parametrise an `AnalogRot` that error is raised:

```python
tests/qadence/test_mitigation.py:134: in <module>
    chain(AnalogRot(duration="t", omega="omega", delta="delta"), AnalogRZ(pi)),
qadence/operations.py:1196: in AnalogRot
    alpha = duration * sympy.sqrt(omega**2 + delta**2) / 1000  # type: ignore [operator]
E   TypeError: unsupported operand type(s) for ** or pow(): 'str' and 'int'
```

### PR DESCRIPTION
**Pull request by Roland-djee**
Closes #242

### CODE REVIEW

**Review by dominikandreasseitz — APPROVED** [review]
can we add a test for this atleast? i guess it went undiscovered because this line was never executed?

**Comment by Roland-djee** [comment]
> can we add a test for this atleast? i guess it went undiscovered because this line was never executed?

Yes, it feels very much like untested functionality 😞

File: qadence/operations.py [hunk 1]
```diff
@@ -1154,7 +1154,7 @@ def AnalogRot(
     q = _cast(QubitSupport, qubit_support)
     if isinstance(duration, str):
         duration = Parameter(duration)
-    alpha = duration * sympy.sqrt(omega**2 + delta**2) / 1000  # type: ignore [operator]
+    alpha = duration * sympy.sqrt(Parameter(omega) ** 2 + Parameter(delta) ** 2) / 1000  # type: ignore [operator]
```
Comment (jpmoutinho) [inline qadence/operations.py:1157]: What happens here if `omega` and `delta` are already passed as `Parameter` ? Is calling `Parameter(Parameter(...))` ok?

I hadn't seen this MR earlier and also added this fix to [#244](https://github.com/pasqal-io/qadence/pull/244) but there I used the `if isinstance(..., str):` logic as done for the `duration` here.

Comment (dominikandreasseitz) [reply qadence/operations.py:1157]: @jpmoutinho its fine

Comment (jpmoutinho) [reply qadence/operations.py:1157]: Ok nice

**Review by nmheim — APPROVED** [review]

File: qadence/operations.py (hunk 1 above)
Comment (Roland-djee) [reply qadence/operations.py:1157]: Nevertheless @jpmoutinho there is what we discussed in the way to test this.

**Review by jpmoutinho — APPROVED** [review]

**Pull request merged**